### ISSUE TITLE
Feature request: Need to be logged in to see status page

### ISSUE THREAD

**Issue by schniepern** [CachetHQ/Cachet#2812]
Hi everyone! :)

In our company we would first only like to make our status page available to logged in users (and be able to turn this off for a later point). I tried to implement this myself without doing too many changes to the source code but I ran into a few problems:
- The way the routing is structured, doing this in one centralized place (like RouteServiceProvider.php) isn't easily possible - it makes the code messy and since I'm not very experienced with Laravel, it would also be kind of unpredictable. Here's what I've done (in RouteServiceProvider.php):
```php
$alwaysAuthenticate = true;
if($alwaysAuthenticate) {
    $router->group(['namespace' => $this->namespace], function (Router $router) {
        $needAuth = ['FeedRoutes', 'StatusPageRoutes', 'SubscribeRoutes'];
        $needAuthBasenames = [];
        foreach (glob(app_path('Http//Routes').'/*.php') as $file) {
            $basename = basename($file, '.php');
            if(in_array($basename, $needAuth)) {
                $needAuthBasenames[] = $basename;
            } else {
                $this->app->make('CachetHQ\\Cachet\\Http\\Routes\\'.$basename)->map($router);
            }
        }
        $router->group(['middleware' => ['web', 'auth']], function (Router $router) use ($needAuthBasenames) {
            foreach($needAuthBasenames as $basename) {
                $this->app->make('CachetHQ\\Cachet\\Http\\Routes\\'.$basename)->map($router);
            }
        });
    });
} else { // original code
    $router->group(['namespace' => $this->namespace], function (Router $router) {
        foreach (glob(app_path('Http//Routes').'/*.php') as $file) {
            $this->app->make('CachetHQ\\Cachet\\Http\\Routes\\'.basename($file, '.php'))->map($router);
        }
    });
}
```
$alwaysAuthenticate would of course later be configurable in the settings.

This code defines a list of additional Routes that need authentication and creates a group with the necessary middleware for only them. It's of course just a messy workaround - it works (as far as I tested, which is to say there might be some bugs a more experienced Laravel developer would know about) but you have to add every *Routes.php file and the 'web' middleware is sometimes added twice (it needs to be added to ensure they handle the request in the right order).

Of course it would be possible to just add the 'auth' middleware in the *Routes.php files under the condition that the feature is enabled, but this would result in a lot of redundant code.

### In conclusion:
- This would be a very interesting and useful feature to implement directly into Cachet
- It isn't easy to find an elegant solution if you want to implement it yourself without messing too much with the source code - a little bit of restructuring on your side would probably do the job! Please correct me if I'm wrong - as I said, I'm an inexperienced Laravel developer, so there might just be something I'm missing.

Thank you very much in advance :)

**Comment by jbrooksuk** [comment]
I like it. Would be very happy to see a PR here.

**Comment by nstapelbroek** [comment]
A PR was just merged that implements this feature. 

@schniepern are you available to do a quick QA review on the 2.4 branch for this feature?

**Comment by jbrooksuk** [comment]
We forgot to close this.
